src/gasman.c

```diff
@@ -1687,6 +1690,10 @@ UInt CollectBags (
     for ( i = 0; i < GlobalBags.nr; i++ )
         MarkBag( *GlobalBags.addr[i] );
 
+    /* allow users to mark their choice of bags */
```

Comment:
this is not used in GAP, and thus somebody who removes dead code might do so.

So perhaps make it explicit that this is there for the benefit of code linking against GAP?

Reply:
This point really is my only objection to this patch: I still think this comment is potentially misleading, or at least not as helpful as it could  be. Perhaps this:

`/* allow installing a custom marking function. This is used for integrating GAP (possibly linked as a shared library) with other code bases which use their own form of garbage collection. For example, with Python (for SageMath) or Julia. */`

This is just meant to convey the gist, I am sure it can be improved a lot.